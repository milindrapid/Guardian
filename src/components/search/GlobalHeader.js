import React from 'react';
import SearchIcon from '../../images/search-icon.svg';
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import styled from 'styled-components';
import PropTypes from 'prop-types';

const SearchBar = styled.div`
    height:4rem;
    top:0;
    left:0;
    right:0;
    display: flex;
    flex-direction: row;
    position: relative;
    padding:0 1rem 0 1.2rem;
    background-color: white;
`;
const SearchInput = styled.input`
    width: 100%;
    padding: 1.4rem 1.3rem 1.4rem 0;
    border: none;
    font-style: normal;
    font-size: 1rem;
    line-height: 1.2rem;
    font-weight: normal;
    font-family:Effra;
    color: #75757b;
    &:focus{
        outline: none;
    }
`;
const SearchMenuIcons = styled.div`
    height:4rem;
    padding-right:1rem;
    ul{
        height:100%;
        display:flex;
        flex-direction:row;
    }
`;
const SearchMenuFontIcon = styled.div`
    font-style:normal
    font-weight: 900;
    align-self: center;
    color: #0d3f5e;
    li{
        font-size: 1.25rem;
        margin-left:.35rem;
    }
    &:hover{
        color: #09809b
    }
`;
const SearchMenuSVGIcon = styled.img`
    width: 1.25rem;
    height: 1.25rem;
    align-self: center;
    margin-left: 1.3rem;
    margin-right: 1.25rem;
    padding: 1px;
`;

const globalHeaderIconList = [
    {
        id: 1,
        name: "Email",
        image: "envelope-open-text"
    },
    {
        id: 2,
        name: "Calender",
        image: "calendar-alt"
    },
    {
        id: 3,
        name: "Activity",
        image: "chart-line"
    },
    {
        id: 4,
        name: "Favorite Tools",
        image: "th"
    }
]


const GlobalHeader = (props) => {
    const { placeHolder, searchCallBack, onSearchMenuIconClick } = props

    const searchResultCallBack = (event) => {
        if (event.which === 13) { //code of enter button is 13
            searchCallBack(event.target.value)
        }
    }

    const onMenuIconClick = (id) => {
        onSearchMenuIconClick(id)
    }

    const getIcons = () => {
        return (
            globalHeaderIconList.map(icon => (
                <SearchMenuFontIcon>
                    <li key={icon.id}>
                        <FontAwesomeIcon
                            icon={icon.image}
                            onClick={() => onMenuIconClick(icon.id)} />
                    </li>
                </SearchMenuFontIcon>
            ))
        )
    }

    return (
        <SearchBar>
            <SearchMenuSVGIcon src={SearchIcon} />
            <SearchInput
                type="text"
                placeholder={placeHolder}
                onKeyPress={searchResultCallBack} />

            <SearchMenuIcons>
                <ul>
                    {getIcons()}
                </ul>
            </SearchMenuIcons>
        </SearchBar>
    )
}

GlobalHeader.propTypes = {
    placeHolder: PropTypes.string.isRequired,
    searchCallBack: PropTypes.func.isRequired,
    onSearchMenuIconClick: PropTypes.func.isRequired
};

export default GlobalHeader;