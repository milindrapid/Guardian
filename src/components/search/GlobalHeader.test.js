import React from "react";
import GlobalHeader from "./GlobalHeader";
import { shallow } from "enzyme";

const globalHeaderIconList = [
    {
        id: 1,
        name: "Email",
        image: "envelope-open-text"
    },
    {
        id: 2,
        name: "Calender",
        image: "calendar-alt"
    },
    {
        id: 3,
        name: "Activity",
        image: "chart-line"
    },
    {
        id: 4,
        name: "Favorite Tools",
        image: "th"
    }
]


describe("GlobalHeader component", () => {
    const props = {
        iconList: globalHeaderIconList,
        placeHolder: "Search Guardian Online..."
    };

    const wrapperComponent = shallow(<GlobalHeader {...props} />);

    it('Global header component should render correclty', () => {
        expect(wrapperComponent).toMatchSnapshot();
    });

    it("should render children components correclty", () => {
        expect(wrapperComponent.find("ul")).toHaveLength(1);
        expect(wrapperComponent.find("li")).toHaveLength(4);
    });
});
