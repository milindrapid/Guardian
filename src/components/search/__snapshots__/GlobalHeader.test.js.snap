// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalHeader component Global header component should render correclty 1`] = `
<styled.div>
  <styled.img
    src="search-icon.svg"
  />
  <styled.input
    onKeyPress={[Function]}
    placeholder="Search Guardian Online..."
    type="text"
  />
  <styled.div>
    <ul>
      <styled.div>
        <li
          key="1"
        >
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon="envelope-open-text"
            inverse={false}
            listItem={false}
            mask={null}
            onClick={[Function]}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          />
        </li>
      </styled.div>
      <styled.div>
        <li
          key="2"
        >
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon="calendar-alt"
            inverse={false}
            listItem={false}
            mask={null}
            onClick={[Function]}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          />
        </li>
      </styled.div>
      <styled.div>
        <li
          key="3"
        >
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon="chart-line"
            inverse={false}
            listItem={false}
            mask={null}
            onClick={[Function]}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          />
        </li>
      </styled.div>
      <styled.div>
        <li
          key="4"
        >
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon="th"
            inverse={false}
            listItem={false}
            mask={null}
            onClick={[Function]}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          />
        </li>
      </styled.div>
    </ul>
  </styled.div>
</styled.div>
`;
